Packs/AzureRiskyUsers/Integrations/AzureRiskyUsers/AzureRiskyUsers.py

```diff
@@ -0,0 +1,395 @@
+
+        return:
+            Response (dict): API response from AzureRiskyUsers.
+        """
+        return self.ms_client.http_request(method='GET',
+                                           url_suffix=f'identityProtection/riskyUsers/{id}')
+
+    def risk_detections_list(self, risk_state: Optional[str], risk_level: Optional[str], limit: int,
+                             skip_token: str = None) -> dict:
+        """
+        Get a list of the Risk Detection objects and their properties.
+
+        Args:
+            risk_state (str): Risk State to retrieve.
+            risk_level (str): Specify to get only results with the same Risk Level.
+            limit (int): Limit of results to retrieve.
+            skip_token (int): Skip token.
+
+        return:
+            Response (dict): API response from AzureRiskyUsers.
+        """
+        params = remove_empty_elements({'$top': limit,
+                                        '$skiptoken': skip_token,
+                                        '$filter': build_query_filter(risk_state, risk_level)})
+
+        return self.ms_client.http_request(method='GET',
+                                           url_suffix="/identityProtection/riskDetections",
+                                           params=params)
+
+    def risk_detection_get(self, id: str) -> dict:
+        """
+        Read the properties and relationships of a riskDetection object.
+
+        Args:
+            id (str): ID of risk detection to retrieve.
+
+        Return:
+            Response (dict): API response from AzureRiskyUsers.
+        """
+        return self.ms_client.http_request(method='GET',
+                                           url_suffix=f'/identityProtection/riskDetections/{id}')
+
+
+def build_query_filter(risk_state: Optional[str], risk_level: Optional[str]) -> Optional[str]:
+    """
+    Build query filter for API call, in order to get filtered results.
+
+    Args:
+        risk_state (str): Wanted risk state for filter.
+        risk_level (str): Wanted risk level for filter.
+
+    Returns:
+        str: Query filter string for API call.
+    """
+    if risk_state and risk_level:
+        return f"riskState eq '{risk_state}' and riskLevel eq '{risk_level}'"
+    elif risk_state:
+        return f"riskState eq '{risk_state}'"
+    elif risk_level:
+        return f"riskLevel eq '{risk_level}'"
+    else:
+        return None
+
+
+def create_event_or_incident_output(item: Dict,
+                                    table_headers: List[str]) -> Dict[str, Optional[Any]]:
+    """
+    Create the complete output dictionary for events or incidents.
+
+    Args:
+        item (dict): A source dictionary from the API response.
+        table_headers (list(str)): The table headers to be used when creating initial data.
+
+    Returns:
+        object_data (dict(str)): The output dictionary.
+    """
+    return remove_empty_elements({field: item.get(field) for field in table_headers})
```

Comment:
So basically this function just removes everything that isn't in the headers + empty elements?
In my opinion, the current name of the function doesn't reflect that.

Also - why is this function necessary?

Reply:
Done